events.go

```diff
@@ -286,26 +286,26 @@ type EventDeploymentFailed struct {
 
 // EventDeploymentInfo describes a 'deployment_info' event.
 type EventDeploymentInfo struct {
-	EventType   string          `json:"eventType"`
-	CurrentStep *DeploymentStep `json:"currentStep"`
-	Timestamp   string          `json:"timestamp"`
-	Plan        *DeploymentPlan `json:"plan"`
+	EventType   string            `json:"eventType"`
+	CurrentStep *DeploymentStepV1 `json:"currentStep"`
```

Comment:
I'm not sure if we should just use `CurrentStep` to replace `DeploymentStepV1` in [unmarshaling deployments](https://github.com/gambol99/go-marathon/blob/2e73c1d824745b3b6374f2d7cc46932f5a596698/deployment.go#L91). I believe a `CurrentStep` placed there would be confusing. Following is my options:

1. Keep the `DeploymentStepV1` as it is and we don't need to change anything more. Because `EventDeploymentInfo.CurrentStep` is surely one kind of deployment step.
2. If option 1 makes some confusion, we can copy `DeploymentStepV1` to a new struct and name it `CurrentStep`. Then we use this new `CurrentStep` in `EventDeploymentInfo.CurrentStep`, and keep `DeploymentStepV1` in [unmarshaling deployments](https://github.com/gambol99/go-marathon/blob/2e73c1d824745b3b6374f2d7cc46932f5a596698/deployment.go#L91). I mentioned the reason at the beginning. (This is what I mentioned in my last comment. )
3. Maybe we could make some refactoring in [unmarshaling deployments](https://github.com/gambol99/go-marathon/blob/2e73c1d824745b3b6374f2d7cc46932f5a596698/deployment.go#L89-L105). Considering `DeploymentStepV1` is later than `DeploymentStep`, we could reverse their place. We could try to unmarshal `DeploymentStepV1` at first. If failed, we then unmarshal `deployment.XXStepsRaw` to `DeploymentStep`. After that, we can name current `DeploymentStepV1` as `DeploymentStep` and current `DeploymentStep` as `DeploymentStepV0` or something else, and make `EventDeploymentInfo.CurrentStep` is type `DeploymentStep` which is `DeploymentStepV1` before. I think this would be more reasonable.